### ISSUE TITLE
NSwag Studio typescript client generate setting request

### ISSUE THREAD

**Issue by Billskate** [RicoSuter/NSwag#2164]
Here is situation, if an operation has many optional parameters, like:
```c#
[HttpGet]
public void Operation(int? a, string b, bool? c) {}
```

the Studio will generate typescript client operation like this:
```typescript
operation(a?: number | undefined, b?: string | undefined, c?: boolean | undefined): Observable;
```

if I only want pass 'c' parameter when I invoke this operation, I have to also pass 'a' and 'b':

```typescript
operation(undefined, undefined, false);
```

if there are many many optional parameters, it will become:
```typescript
operation(undefined, undefined, undefined, undefined, undefined, undefined, false);
```
so, my point is, can Studio provide a setting that wrap all parameters into a query object, it will become more friendly when invoke an operation:
```typescript
// operation declaration
operation(query: QueryPamaeter): Observable;

// query object declaration
interface QueryPamaeter {
  a?: number | undefined;
  b?: string | undefined;
  c?: boolean | undefined;
}

//invoke
operation({
  c: false
});
```

**Comment by akamyshanov** [comment]
https://github.com/RicoSuter/NSwag/issues/950

**Comment by Billskate** [comment]
closed, see: #950
